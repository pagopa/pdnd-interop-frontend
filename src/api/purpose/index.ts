export * from './purpose.queries'
export * from './purpose.downloads'
export * from './purpose.mutations'
export * from './purpose.services'
