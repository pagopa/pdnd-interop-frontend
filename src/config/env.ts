import type { ExtendedWindow, PagoPAEnvVars } from '@/types/common.types'

const PAGOPA_ENV = (window as unknown as ExtendedWindow).pagopa_env

export const isDevelopment = !!(import.meta.env.MODE === 'development')
export const isProduction = !!(import.meta.env.MODE === 'production')
export const isTest = !!(import.meta.env.MODE === 'test')

if (!PAGOPA_ENV && !isTest) {
  console.warn('pagopa_env not available.')
}

const DEV_API_HOST_URL = import.meta.env.REACT_APP_API_HOST
const DEV_SELFCARE_LOGIN_URL = import.meta.env.REACT_APP_SELFCARE_LOGIN_URL
const DEV_INTEROP_RESOURCES_BASE_URL = import.meta.env.REACT_APP_INTEROP_RESOURCES_BASE_URL
const DEV_ONETRUST_DOMAIN_SCRIPT_ID = import.meta.env.REACT_APP_ONETRUST_DOMAIN_SCRIPT_ID

export const FE_LOGIN_URL = (
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.SELFCARE_LOGIN_URL : DEV_SELFCARE_LOGIN_URL
) as string
export const PUBLIC_URL = import.meta.env.PUBLIC_URL || '/ui'
export const FE_URL = `${window.location.origin}${PUBLIC_URL}`

export const INTEROP_RESOURCES_BASE_URL =
  isProduction && PAGOPA_ENV
    ? PAGOPA_ENV.INTEROP_RESOURCES_BASE_URL
    : DEV_INTEROP_RESOURCES_BASE_URL
export const ONETRUST_DOMAIN_SCRIPT_ID = (
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.ONETRUST_DOMAIN_SCRIPT_ID : DEV_ONETRUST_DOMAIN_SCRIPT_ID
) as string
export const MIXPANEL_PROJECT_ID = isProduction && PAGOPA_ENV ? PAGOPA_ENV.MIXPANEL_PROJECT_ID : ''

export const TEMP_USER_WHITELIST_URL = `${INTEROP_RESOURCES_BASE_URL}/temp-whitelist.json`

function getEnvVar(varName: keyof PagoPAEnvVars, devVarName: string): string {
  return isProduction && PAGOPA_ENV && PAGOPA_ENV
    ? PAGOPA_ENV[varName]
    : `${DEV_API_HOST_URL}/${devVarName}`
}

const SERVICE_VERSION = import.meta.env.REACT_APP_SERVICE_VERSION
export const API_GATEWAY_INTEFACE_URL = getEnvVar(
  'API_GATEWAY_INTEFACE_URL',
  'swagger/docs/interface-specification.yml'
)
export const BACKEND_FOR_FRONTEND_URL = getEnvVar(
  'BACKEND_FOR_FRONTEND_URL',
  `backend-for-frontend/${SERVICE_VERSION}`
)
export const AUTHORIZATION_PROCESS_URL = getEnvVar(
  'AUTHORIZATION_PROCESS_URL',
  `authorization-process/${SERVICE_VERSION}`
)
export const AUTHORIZATION_SERVER_ACCESS_TOKEN_URL = getEnvVar(
  'AUTHORIZATION_SERVER_TOKEN_CREATION_URL',
  'authorization-server/token.oauth2'
)
export const CLIENT_ASSERTION_JWT_AUDIENCE =
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.CLIENT_ASSERTION_JWT_AUDIENCE : ''

function getWellKnownUrls(wellKnownUrls: string | undefined) {
  return wellKnownUrls?.split(',').filter((url) => !!url) || []
}

export const WELL_KNOWN_URLS =
  isProduction && PAGOPA_ENV ? getWellKnownUrls(PAGOPA_ENV.WELL_KNOWN_URLS) : ['#']

export const SELFCARE_BASE_URL =
  isProduction && PAGOPA_ENV ? PAGOPA_ENV.SELFCARE_BASE_URL : 'https://uat.selfcare.pagopa.it'

export const STAGE = PAGOPA_ENV?.STAGE ?? 'DEV'

export const SELFCARE_INTEROP_PROD_ID = `prod-interop${STAGE === 'TEST' ? '-coll' : ''}`
