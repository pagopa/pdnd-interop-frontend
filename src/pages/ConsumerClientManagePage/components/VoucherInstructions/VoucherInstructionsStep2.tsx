import { InformationContainer, SectionContainer } from '@/components/layout/containers'
import { StepActions } from '@/components/shared/StepActions'
import { AUTHORIZATION_SERVER_ACCESS_TOKEN_URL, FE_URL } from '@/config/env'
import { Divider, Stack, Typography } from '@mui/material'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useClientKind } from '@/hooks/useClientKind'
import type { VoucherInstructionsStepProps } from '../../types/voucher-instructions.types'
import { CodeSnippetPreview } from './CodeSnipperPreview'
import { InlineClipboard } from '@/components/shared/InlineClipboard'

const CLIENT_ASSERTION_TYPE = 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer'
const GRANT_TYPE = 'client_credentials'

export const VoucherInstructionsStep2: React.FC<VoucherInstructionsStepProps> = ({
  back,
  forward,
  clientId,
}) => {
  const { t } = useTranslation('voucher')
  const clientKind = useClientKind()

  return (
    <SectionContainer>
      <Typography component="h2" variant="h5">
        {t('step2.title')}
      </Typography>
      <Typography sx={{ mt: 1 }} component="p" variant="body1" color="text.secondary">
        {t(`step2.${clientKind === 'CONSUMER' ? 'consumerDescription' : 'apiDescription.message'}`)}
      </Typography>

      <InformationContainer sx={{ mt: 4 }} label={t('step2.authEndpoint.label')}>
        <InlineClipboard
          textToCopy={AUTHORIZATION_SERVER_ACCESS_TOKEN_URL}
          successFeedbackText={t('step2.authEndpoint.copySuccessFeedbackText')}
        />
      </InformationContainer>

      <Divider sx={{ mt: 4 }} />

      <Typography sx={{ my: 4 }} component="h2" variant="h6">
        {t('step2.requestBody.title')}
      </Typography>

      <Stack spacing={4}>
        <InformationContainer label={t('step2.requestBody.clientIdField.label')}>
          <InlineClipboard
            textToCopy={clientId}
            successFeedbackText={t('step2.requestBody.clientIdField.copySuccessFeedbackText')}
          />
        </InformationContainer>

        <InformationContainer label={t('step2.requestBody.clientAssertionField.label')}>
          {t('step2.requestBody.clientAssertionField.suggestionLabel')}
        </InformationContainer>

        <InformationContainer
          label={t('step2.requestBody.clientAssertionTypeField.label')}
          labelDescription={t('step2.requestBody.clientAssertionTypeField.description')}
        >
          <InlineClipboard
            textToCopy={CLIENT_ASSERTION_TYPE}
            successFeedbackText={t(
              'step2.requestBody.clientAssertionTypeField.copySuccessFeedbackText'
            )}
          />
        </InformationContainer>

        <InformationContainer
          label={t('step2.requestBody.grantTypeField.label')}
          labelDescription={t('step2.requestBody.grantTypeField.description')}
        >
          <InlineClipboard
            textToCopy={GRANT_TYPE}
            successFeedbackText={t('step2.requestBody.grantTypeField.copySuccessFeedbackText')}
          />
        </InformationContainer>
      </Stack>

      <Divider sx={{ mt: 4 }} />

      <Typography sx={{ mt: 3 }} component="h3" variant="h6">
        {t('step2.voucherScript.title')}
      </Typography>
      <Typography sx={{ mt: 1 }} component="p" variant="body2" color="text.secondary">
        {t('step2.voucherScript.description')}
      </Typography>

      <CodeSnippetPreview
        sx={{ mt: 2 }}
        title={t('step2.voucherScript.exampleLabel')}
        activeLang="curl"
        entries={[{ url: `${FE_URL}/data/it/session_token_curl.txt`, value: 'curl' }]}
        scriptSubstitutionValues={{
          AUTHORIZATION_SERVER_ACCESS_TOKEN_URL,
          CLIENT_ID: clientId,
          CLIENT_ASSERTION_TYPE: CLIENT_ASSERTION_TYPE,
          GRANT_TYPE: GRANT_TYPE,
        }}
      />

      <StepActions
        back={{ label: t('backBtn'), type: 'button', onClick: back }}
        forward={{ label: t('proceedBtn'), type: 'button', onClick: forward }}
      />
    </SectionContainer>
  )
}
