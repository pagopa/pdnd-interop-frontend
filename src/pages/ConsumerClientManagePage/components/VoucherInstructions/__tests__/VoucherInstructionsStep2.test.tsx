import React from 'react'
import type { VoucherInstructionsStepProps } from '@/pages/ConsumerClientManagePage/types/voucher-instructions.types'
import {
  mockUseClientKind,
  renderWithApplicationContext,
  setupQueryServer,
} from '@/utils/testing.utils'
import { createMockPublicKey } from '__mocks__/data/key.mocks'
import { createMockPurpose } from '__mocks__/data/purpose.mocks'
import { vi } from 'vitest'
import { FE_URL } from '@/config/env'
import sessionTokenCurl from '../../../../../../public/data/it/session_token_curl.txt'
import { VoucherInstructionsStep2 } from '../VoucherInstructionsStep2'

const queryServer = setupQueryServer([
  {
    url: `${FE_URL}/data/it/session_token_curl.txt`,
    result: sessionTokenCurl,
  },
])

beforeAll(() => queryServer.listen())
afterEach(() => queryServer.resetHandlers())
afterAll(() => queryServer.close())

const commonProps: VoucherInstructionsStepProps = {
  clientId: 'test-client-id',
  clientKeys: {
    keys: [
      createMockPublicKey({ keyId: '1', name: 'key1' }),
      createMockPublicKey({ keyId: '2', name: 'key2' }),
    ],
  },
  purpose: createMockPurpose(),
  purposeId: 'test-purpose-id',
  forward: vi.fn(),
  back: vi.fn(),
}

describe('VoucherInstructionsStep2', () => {
  it('should match snapshot (API)', () => {
    mockUseClientKind('API')
    const screen = renderWithApplicationContext(<VoucherInstructionsStep2 {...commonProps} />, {
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot()
  })

  it('should match snapshot (CONSUMER)', () => {
    mockUseClientKind('CONSUMER')
    const screen = renderWithApplicationContext(<VoucherInstructionsStep2 {...commonProps} />, {
      withRouterContext: true,
    })
    expect(screen.baseElement).toMatchSnapshot()
  })
})
