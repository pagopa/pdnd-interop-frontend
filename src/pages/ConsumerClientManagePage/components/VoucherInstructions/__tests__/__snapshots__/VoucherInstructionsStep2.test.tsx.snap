// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VoucherInstructionsStep2 > should match snapshot (API) 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step2.apiDescription.message
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-1tixoof-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              step2.authEndpoint.label
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              https://test.it//authorization-server/token.oauth2
              <button
                aria-label="Copia"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                role="button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="ContentCopyIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.requestBody.title
        </h2>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-ramisn-MuiStack-root"
        >
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientIdField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                test-client-id
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientAssertionField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                step2.requestBody.clientAssertionField.suggestionLabel
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientAssertionTypeField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step2.requestBody.clientAssertionTypeField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                urn:ietf:params:oauth:client-assertion-type:jwt-bearer
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.grantTypeField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step2.requestBody.grantTypeField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                client_credentials
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.voucherScript.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step2.voucherScript.description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiBox-root css-9lwr4s"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-xomfih-MuiTypography-root"
          >
            step2.voucherScript.exampleLabel
          </span>
          <div
            class="MuiBox-root css-79elbk"
          >
            <div
              class="MuiBox-root css-1azb3nq"
            >
              <div
                class="MuiBox-root css-tx9fw1"
              />
            </div>
            <div
              class="MuiBox-root css-1f1zykm"
            />
          </div>
        </div>
        <div
          class="css-9v4aab-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            backBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            proceedBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`VoucherInstructionsStep2 > should match snapshot (CONSUMER) 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step2.consumerDescription
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-1tixoof-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              step2.authEndpoint.label
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              https://test.it//authorization-server/token.oauth2
              <button
                aria-label="Copia"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                role="button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="ContentCopyIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.requestBody.title
        </h2>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="css-ramisn-MuiStack-root"
        >
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientIdField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                test-client-id
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientAssertionField.label
              </p>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                step2.requestBody.clientAssertionField.suggestionLabel
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.clientAssertionTypeField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step2.requestBody.clientAssertionTypeField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                urn:ietf:params:oauth:client-assertion-type:jwt-bearer
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
          <div
            class="css-19qjufr-MuiStack-root"
          >
            <div
              class="MuiBox-root css-iuqs4u"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
              >
                step2.requestBody.grantTypeField.label
              </p>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-159wgzx-MuiDivider-root"
              />
              <span
                class="MuiTypography-root MuiTypography-caption css-g171c3-MuiTypography-root"
              >
                step2.requestBody.grantTypeField.description
              </span>
            </div>
            <div
              class="MuiBox-root css-1rr4qq7"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                client_credentials
                <button
                  aria-label="Copia"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-vqczhe-MuiButtonBase-root-MuiIconButton-root"
                  data-mui-internal-clone-element="true"
                  role="button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="ContentCopyIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1w5x330-MuiPaper-root"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
        >
          step2.voucherScript.title
        </h2>
        <span
          class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
        >
          step2.voucherScript.description
        </span>
      </div>
      <div
        class="MuiBox-root css-1yuhvjn"
      >
        <div
          class="MuiBox-root css-9lwr4s"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-xomfih-MuiTypography-root"
          >
            step2.voucherScript.exampleLabel
          </span>
          <div
            class="MuiBox-root css-79elbk"
          >
            <div
              class="MuiBox-root css-1azb3nq"
            >
              <div
                class="MuiBox-root css-tx9fw1"
              />
            </div>
            <div
              class="MuiBox-root css-1f1zykm"
            />
          </div>
        </div>
        <div
          class="css-9v4aab-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            backBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            proceedBtn
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
